mountpoint-s3/tests/fuse_tests/write_test.rs

```diff
@@ -435,6 +437,65 @@ where
     assert_eq!(err.raw_os_error(), Some(libc::ENOENT));
 }
 
+fn write_with_storage_class_test<F>(creator_fn: F, storage_class: &str)
+where
+    F: FnOnce(&str, TestSessionConfig) -> (TempDir, BackgroundSession, TestClientBox),
+{
+    const KEY: &str = "new.txt";
+
+    let filesystem_config: S3FilesystemConfig = S3FilesystemConfig {
+        storage_class: Some(storage_class.to_owned()),
+        ..Default::default()
+    };
+
+    let config = TestSessionConfig {
+        filesystem_config,
+        ..Default::default()
+    };
+    let (mount_point, _session, test_client) = creator_fn("write_with_storage_class_test", config);
+
+    let path = mount_point.path().join(KEY);
+
+    let mut f = open_for_write(&path, false).unwrap();
+
+    let data = [0xaa; 16];
+    f.write_all(&data).unwrap();
+    f.sync_all().unwrap();
+    drop(f);
```

Comment:
Could you use `write_file` here as well? Provided you add the `sync_all` call to it.